src/Process/Runner/MutationTestingRunner.php

```diff
@@ -89,11 +89,10 @@ public function run(iterable $mutations, string $testFrameworkExtraOptions): voi
             ->cast(fn (Mutant $mutant) => $this->mutantToContainer($mutant, $testFrameworkExtraOptions))
         ;
 
-        foreach ($this->processRunner->run($processContainers) as $mutantProcessContainer) {
-            $this->eventDispatcher->dispatch(new MutantProcessWasFinished(
-                $mutantProcessContainer->getCurrent()->getMutantExecutionResult(),
-            ));
-        }
+        take($this->processRunner->run($processContainers))
+            ->cast(self::containerToFinishedEvent(...))
+            ->each($this->eventDispatcher->dispatch(...))
+        ;
```

Comment:
So we can test `containerToFinishedEvent` without having to involve anything else.